components/core/src/streaming_archive/writer/Archive.hpp

```diff
@@ -8,6 +8,7 @@
 #include <unordered_set>
 #include <vector>
 #include <filesystem>
+#include <optional>
```

Comment:
Since you are adding a new header, can you help to sort the headers in the alphabet order? :D